WolvenKit.App/ViewModels/Shell/ChunkViewModel.cs

```diff
@@ -28,65 +31,75 @@ namespace WolvenKit.ViewModels.Shell
-                .ToPropertyEx(this, x => x.Descriptor, deferSubscription: true);
-            dataObserver
-                .Select(_ => CalculateIsDefault())
-                .ToPropertyEx(this, x => x.IsDefault, deferSubscription: true);
-            if (Parent != null)
-            {
-                this.WhenAnyValue(x => x.Data, x => x.Parent.IsExpanded, x => x.ForceLoadProperties)
-                    .Where(_ => Parent.IsExpanded || ForceLoadProperties)
-                    .Select(_ => CalculateProperties())
-                    .ToPropertyEx(this, x => x.Properties, deferSubscription: true);
-            }
-            else
-            {
-                this.WhenAnyValue(x => x.Data, x => x.ForceLoadProperties)
-                    .Select(_ => CalculateProperties())
-                    .ToPropertyEx(this, x => x.Properties, deferSubscription: true);
-            }
-            this.WhenAnyValue(x => x.Properties)
-                .Select((_) =>
+            Properties.ToObservableChangeSet()
+                .Subscribe(x =>
                 {
+                    TVProperties.Clear();
+                    DisplayProperties.Clear();
+
                     if (Properties == null || Properties.Count == 0)
                     {
-                        return SelfList;
+                        DisplayProperties.AddRange(SelfList);
                     }
                     else
                     {
-                        return Properties;
+                        if (Data is not IRedArray && Data is not IRedBufferWrapper)
+                        {
+                            TVProperties.AddRange(Properties);
+                            DisplayProperties.AddRange(Properties);
+                        }
+                        else
+                        {
+                            var settings = Locator.Current.GetService<ISettingsManager>();
+                            if (settings.TreeViewGroups && settings.TreeViewGroupSize > 1)
+                            {
+                                TVProperties.AddRange(SplitProperties(Properties, (int)settings.TreeViewGroupSize));
+                            }
+                            else
+                            {
+                                TVProperties.AddRange(Properties);
+                            }
+
+                            
+                            DisplayProperties.AddRange(SelfList);
+                        }
                     }
-                }).ToPropertyEx(this, x => x.DisplayProperties, deferSubscription: true);
 
-            //dataObserver
-            //    .Select(x => CalculatePropertyGridData())
-            //    .ToPropertyEx(this, x => x.PropertyGridData);
-            dataObserver
+                    CalculateDescriptor();
+                });
+
+            this.WhenAnyValue(x => x.IsSelected)
+                .Where(x => IsSelected && !_propertiesLoaded)
+                .Subscribe(x => CalculateProperties());
```

Comment:
Minor naming: all the `Calculate*` are now effectively `Set*`/`Update*` or something that indicates they’re side-effecting?